hooks/post_gen_project.py

```diff
@@ -0,0 +1,76 @@
+"""Post project generation hook."""
+
+import shlex
+import subprocess
+import sys
+
+_EXIT_FAILURE = 1
+_EXIT_SUCCESS = 0
+
+
+def main(initialise_git_repository: str) -> int:
+    """
+    Create a git repository on generation of the project.
+
+    Args:
+        initialise_git_repository: Whether to initialise the repo
+
+    Returns:
+        The return code of the process
+    """
+    if initialise_git_repository == "True":
+        try:
+            # initialise git repo
+            subprocess.run(shlex.split("git init"), check=True)  # noqa: S603
+            # old versions of git still default to `master`
+            subprocess.run(
+                shlex.split("git branch -M main"),  # noqa: S603
+                check=True,
+                capture_output=True,
+            )
+        except FileNotFoundError:
+            # cannot find git
+            print("git isn't installed")  # noqa: T201
```

Comment:
T201 sucks.

Reply:
Yeah it's a bit weird. I don't really want to use something else (like `logging`/`pprint` etc. I'll turn off `T201` for the hooks directory.